src/m365/entra/commands/group/group-set.spec.ts

```diff
@@ -0,0 +1,367 @@
+    const actual = await command.validate({ options: { displayName: 'Cli group', ownerIds: ownerIds.join(',') } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation if ownerUserNames contains invalid user principal name', async () => {
+    const ownerUserNames = ['john.doe@contoso.com', 'foo'];
+    const actual = await command.validate({ options: { displayName: 'Cli group', ownerUserNames: ownerUserNames.join(',') } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation if memberIds contains invalid GUID', async () => {
+    const memberIds = ['7167b488-1ffb-43f1-9547-35969469bada', 'foo'];
+    const actual = await command.validate({ options: { displayName: 'Cli group', memberIds: memberIds.join(',') } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation if memberUserNames contains invalid user principal name', async () => {
+    const memberUserNames = ['john.doe@contoso.com', 'foo'];
+    const actual = await command.validate({ options: { displayName: 'Cli group', memberUserNames: memberUserNames.join(',') } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation if visibility contains invalid value', async () => {
+    const actual = await command.validate({ options: { displayName: 'Cli group', visibility: 'foo' } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('passes validation if id is a valid GUID', async () => {
+    const actual = await command.validate({ options: { id: groupId, newDisplayName: 'Cli group' } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('fails validation if id is not a valid GUID', async () => {
+    const actual = await command.validate({ options: { id: 'foo', newDisplayName: 'Cli group' } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('passes validation when all required parameters are valid with ids', async () => {
+    const actual = await command.validate({ options: { displayName: 'Cli group', ownerIds: userIds.join(','), memberIds: userIds.join(',') } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('passes validation when all required parameters are valid with user names', async () => {
+    const actual = await command.validate({ options: { displayName: 'Cli group', ownerUserNames: userUpns.join(','), memberUserNames: userUpns.join(',') } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('fails validation if no options to be updated is specified', async () => {
+    const actual = await command.validate({ options: { id: groupId } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('successfully updates group specified by id', async () => {
+    sinon.stub(request, 'get').resolves({ value: [] });
+    const patchRequestStub = sinon.stub(request, 'patch').resolves();
```

Comment:
Let's also check for the right API URL.

Reply:
I've removed the check for URL based on this

https://github.com/pnp/cli-microsoft365/pull/5865#discussion_r1676476390

> For this test, we don't really care what the URL of the API is. The only thing we want to test is that we correctly handle errors, no mather what the URL is. So just like your last test, let's just resolve or reject directly whatever URL is passed.

Reply:
I've set this comment on a test that validates if we correctly log an error when the API is failing. In that case, we don't really care if the URL is correct. In this case we're validating if the command is actually working.

Reply:
ok, clear